src/components/App.tsx

```diff
@@ -16,15 +17,11 @@ type ErrorState = {
   errorMessage: string;
 };
 
-type SuccessState = {
-  isLoading: false;
-  networkErrorOccurred: false;
-  headerText: string;
+type State = (LoadingState | ErrorState) & {
```

Comment:
There is no way to represent a success state that has `isLoading: false` and `networkErrorOccurred: false` since it can either be loading or error.